tools/nimgrep.nim

```diff
@@ -118,7 +121,7 @@ proc highlight(s, match, repl: string, t: tuple[first, last: int],
     stdout.write("\n")
     stdout.flushFile()
 
-proc processFile(filename: string) =
+proc processFile(pattern; filename: string) =
```

Comment:
Type-less proc parameters are deprecated.

Reply:
He didn't use them

Reply:
@dom96: this is using a `using` statement for the type: https://github.com/nim-lang/Nim/pull/6983/files#diff-0366ea3274bf3dcbd7108a09fafaf7b6R50